cpp/ql/test/library-tests/dataflow/taint-tests/arrayassignment.cpp

```diff
@@ -53,8 +53,8 @@ void test_myint_member_assignment()
 
 	mi.i = source();
 
-	sink(mi); // $ ir MISSING: ast
-	sink(mi.get()); // $ ast,ir
+	sink(mi); // $ MISSING: ast,ir
+	sink(mi.get()); // $ ast MISSING: ir
```

Comment:
Is there anything we can do to restore this result? Would it have worked if pointers were used instead of references?

Reply:
I'll take a look at it. I'm sure we can restore it, but I'll probably create an issue for it to not add even more code that needs to be reviewed.

Reply:
The underlying issue is that, after doing the store step for `mi.i = source();` and we end up with flow on the node `VariableAddress[mi] [post update]`, we transfer flow to the read side effect on `mi.get()` (since that is the next load of `mi` following the store to `mi`). But in this case, we actually wanted flow to the `this` argument (and not `*this`) since the read inside `get` looks like:
```
r9_10(glval<unknown>) = VariableAddress[#this]       :
r9_11(MyInt *)        = Load[#this]                  : &:r9_10, m9_6
r9_12(glval<int>)     = FieldAddress[i]              : r9_11
r0_1(int &)           = CopyValue                    : r9_12
m0_2(int &)           = Store[#return]               : &:r9_9, r0_1
```
and we correctly infer the read step from `VariableAddress[#this]` that reads `i`.

Now, this _should_ have been handled by 5dbaea8b52b2eaca2acd71ea4fcee125447538c6, but for some reason it isn't. I _think_ it's not handled because the code returns a reference, and thus the code is looking for a load that's not there. I'll create an issue that describes this in more detail, and we can revisit it later.

One quick fix to this is to delete the ugly hack in 5dbaea8b52b2eaca2acd71ea4fcee125447538c6 and accept object-pointer conflation on call-boundaries, by transfering flow to the `this` argument (instead of `*this`) following store steps. All this should be a one-line change to the `CallInstruction` case in `flowOutOfAddressStep`.